test/distributed/fsdp/test_fsdp_misc.py

```diff
@@ -86,6 +87,87 @@ def forward(self, x):
         # https://github.com/pytorch/pytorch/issues/83107 and
         # https://github.com/pytorch/pytorch/issues/83129
 
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_not_all_outputs_used_in_loss(self):
+
+        class MyModule(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.lin1 = nn.Linear(4, 4)
+                self.lin2 = nn.Linear(4, 4)
+
+            def forward(self, x):
+                a = self.lin1(x)
+                b = self.lin2(x)
+                return (a, b)
+
+        def _check_resharded(fsdp_module):
+            for param in fsdp_module.params:
+                if param._is_sharded:
+                    full_param = param._full_param_padded
+                    self.assertEqual(full_param.storage().size(), 0)
+
+                self.assertEqual(
+                    param.data.data_ptr(),
```

Comment:
nit: I think there may be no need to call `.data` here.
```suggestion
                    param.data_ptr(),
```